### ISSUE TITLE
S3 throws a duplicate error when fetching tags for an s3 object with multiple tags

### ISSUE THREAD

**Issue by feikesteenbergen** [durch/rust-s3#178]
**Describe the bug**
S3 throws a duplicate error when fetching tags for an s3 object with multiple tags

**To Reproduce**
- create s3 object

```rust
use s3::{creds::Credentials, Bucket, Region};

fn main() {
    let domain = "www.example.com";
    let credentials = Credentials::default().unwrap();
    let bucket = Bucket::new(domain, Region::UsEast1, credentials).unwrap();

    println!("{:#?}", bucket.get_object_tagging_blocking("ab.tgz"));
}
```

## a single tag
- Add the tag `tag1=value1` to the s3 object
- This is fine, expected behavior
```console
[...]
Ok(
    (
        Some(
            Tagging {
                tag_set: [
                    Tag {
                        kvpair: KVPair {
                            key: "tag1",
                            value: "value1",
                        },
                    },
                ],
            },
        ),
        200,
    ),
)
```

## two tags
- Add the tag `tag1=value1` to the s3 object
- Add the tag `tag2=value2` to the s3 object
- This is not fine, I expect to be able to receive something I can iterate over
```console
[...]
Err(
    S3Error {
        description: Some(
            "custom: duplicate field `Tag`",
        ),
        data: None,
        source: Some(
            Xml(
                Custom {
                    field: "duplicate field `Tag`",
                },
            ),
        ),
    },
)
```

**Expected behavior**
I would expect to be able to iterate over the tags. Instead I get a duplicate error.

**Environment**
 - Rust version: `1.50`
 - lib version [e.g. `0.26.4`]

**Comment by durch** [comment]
@feikesteenbergen thanks for raising this, and sorry for the wait. This indeed does not look right, let me see what I can do

**Comment by durch** [comment]
So this comes down to xml and serde and Rust. All in all a pretty disappointing story. TLDR serde can't really handle XML, I've switched to DOM for xml parsing and so far so good, fix will be published as `0.27.0-rc3`

**Comment by feikesteenbergen** [comment]
@durch thanks for fixing this and working on this crate.
